recipes/jemalloc/all/conanfile.py

```diff
@@ -52,8 +52,10 @@ def config_options(self):
             del self.options.fPIC
 
     def configure(self):
-        if self.settings.compiler.get_safe("libcxx") == "libc++":
-            raise ConanInvalidConfiguration("libc++ is missing a mutex implementation.  Remove this when it is added")
+        if self.settings.compiler.get_safe("libcxx") == "libc++" and \
+                self.settings.compiler == "clang" and \
```

Comment:
I don't think this check is necessary but it makes more sense when we have the next line.

Reply:
I don't feel happy about this, because the recipe has a `enable_cxx` option.
Later in the `configure` option, `libcxx` is removed from `self.settings.compiler`.

Does the build script not include any c++ sources when settings `enable_cxx=False`?

The `ConanInvalidConfiguration` should not be thrown when `enable_cxx` is False, I think.

Reply:
If enable_cxx is False, then cpp file jemalloc_cpp.cpp is not build and this is pure-c project.
I will add an if, so the build will work in the case of old clang and enable_cxx=False

Reply:
Checked, now build successfully with old clang and enable_cxx=False

Reply:
Thanks, so older clangs will be barred when `enable_cxx` is True?

Reply:
And when libc++ is used as a standard library.

Reply:
Great fix!
Nice to have this solved!